tf2onnx/tfonnx.py

```diff
@@ -2280,6 +2285,47 @@ def rewrite_incomplete_type_support_rs6(g, ops):
     return rewrite_incomplete_type_support(g, ops, ["Div", "ReduceSum", "Slice", "Split", "Tile", "Transpose"])
 
 
+def rewrite_conv2d_with_pad(g, ops):
+    pattern = \
+        OpTypePattern("Conv2D", name="conv", inputs=[
+            OpTypePattern("Pad", name="pad"),
+            OpTypePattern("*")
+        ])
+    matcher = GraphMatcher(pattern)
+    match_results = list(matcher.match_ops(ops))
+    for match in match_results:
+        conv = match.get_op("conv")
+        pad = match.get_op("pad")
+        paddings = pad.inputs[1]
+
+        if not paddings.is_const():
+            return ops
+        mode = pad.get_attr("mode")
+        if mode:
+            mode = mode.s.decode("utf-8").lower()
+        if mode not in [None, "constant"] or len(pad.input) >= 3:
+            return ops
+        # Conv2D already has a pad
+        if conv.get_attr("padding") == "SAME":
+            return ops
+
+        log.debug("merge pad [%s] into conv [%s]", pad.name, conv.name)
+        paddings_val = np.array(paddings.get_tensor_value())
+        # can't pad on batch or channel dimensions
+        if np.any(paddings_val[0]) or np.any(paddings_val[3]):
+            return ops
+        paddings_val = paddings_val[1:3]
+        paddings_val = paddings_val.transpose().flatten()
+        g.replace_input(conv, conv.input[0], pad.input[0])
+        # convert Conv2D
```

Comment:
if you want to do conversion here, I think you should also consider opset check.

Reply:
shall we consider to do the fusion in the later rewriter? then we won't have the trick to call mapping function in this stage.

Reply:
or it can even be part of the optimizer.

Reply:
Have thought of this. but then we will need to handle the case in which converted Pad and Conv nodes are separated by Cast and Transpose nodes inserted by the converter.

Reply:
> if you want to do conversion here, I think you should also consider opset check.

Talk with @pengwa offline. We decide not to check OPSET here, as `Conv` op has no change since OPSET 1.
But we find many pre-rewriters create ONNX nodes without doing this check. It will screw up if we use nodes of the wrong version in these rewriters.

Reply:
Not checking sounds correct. I guess if in a future opset something changes we'll depend on unittests to remind us.